### PR TITLE
Fix sonar

### PR DESCRIPTION
**Pull request by npetzall**
This will analyze correctly by sonar.

Extracted the Exception class that caused the issue, verified by using local sonarqube.

Switch to maven profile.

So you can easily analyze locally with for instance sonarqube docker image.

./mvnw -P sonar clean verify

### CODE REVIEW

**Comment by npetzall** [comment]
@rafalkasa should make master green again.

File: travis.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ elif [ "$TRAVIS_PULL_REQUEST" != "false" ] && [ -n "${GITHUB_TOKEN-}" ]; then
 	# http://docs.travis-ci.com/user/pull-requests/#Security-Restrictions-when-testing-Pull-Requests
 	# That's why the analysis does not need to be executed if the variable GITHUB_TOKEN is not defined.
 	echo "Build and analyze internal pull request..."
-	mvn clean verify sonar:sonar -B -e -V \
+	mvn -P sonar clean verifygit  -e -V \
```
Comment (rafalkasa) [inline travis.sh]: @npetzall could you check this line, I can't find Maven goal `verifygit `?

Comment (npetzall) [reply travis.sh]: That git shouldn't be there.

Comment (npetzall) [reply travis.sh]: I'm blaming my son he was sitting in my lap at the time.

Comment (rafalkasa) [reply travis.sh]: haha good :smile: I know how it is coding with child on the knees

**Comment by npetzall** [comment]
Sonar clouds dumped the Sonar Java to i think 5.1.0 which introduced the problem.

**Comment by rafalkasa** [comment]
@npetzall Thank you once again for fixed sonar issue.

**Pull request merged**